sound/soc/sof/pm.c

```diff
@@ -93,6 +93,10 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 	if (!sof_ops(sdev)->resume || !sof_ops(sdev)->runtime_resume)
 		return 0;
 
+	/* DSP was never successfully started, nothing to resume */
+	if (sdev->fw_state == SOF_FW_BOOT_NOT_STARTED && sdev->first_boot)
```

Comment:
@kv2019i resume would be called if we ever suspended successfully. And in this case, the fw_state is always NOT_STARTED. So the first check is redundant I think.

Reply:
@ranj063 Ack, you are right. Replacing "first_boot" with a new fw_state could clarify this, but this would be pretty invasive. I'll change the code.